core/CMakeLists.txt

```diff
@@ -94,72 +125,40 @@ endif()
 
 add_subdirectory(base)
 
+target_sources(Core PRIVATE
+  $<TARGET_OBJECTS:Base>
+  $<TARGET_OBJECTS:Clib>
+  $<TARGET_OBJECTS:Cont>
+  $<TARGET_OBJECTS:Foundation>
+  $<TARGET_OBJECTS:Lzma>
+  $<TARGET_OBJECTS:Lz4>
+  $<TARGET_OBJECTS:Zip>
+  $<TARGET_OBJECTS:Zstd>
+  $<TARGET_OBJECTS:Meta>
+  $<TARGET_OBJECTS:GuiCore>
+  $<TARGET_OBJECTS:TextInput>
+)
+
 if(UNIX)
   add_subdirectory(unix)
-  set(unix_objects $<TARGET_OBJECTS:Unix>)
+  target_sources(Core PRIVATE $<TARGET_OBJECTS:Unix>)
 endif()
 if(WIN32)
   add_subdirectory(winnt)
-  set(winnt_objects $<TARGET_OBJECTS:Winnt>)
+  target_sources(Core PRIVATE $<TARGET_OBJECTS:Winnt>)
 endif()
 if(cocoa)
   add_subdirectory(macosx)
-  set(macosx_objects $<TARGET_OBJECTS:Macosx>)
+  target_sources(Core PRIVATE $<TARGET_OBJECTS:Macosx>)
 endif()
 
 if(CMAKE_CXX_STANDARD GREATER 11)
   set(dict_v7dirs base/v7/inc)
 endif()
 
-#---G__Core--------------------------------------------------------------------
-
-# Uses includes path as defined by Core module.
-
-set(objectlibs $<TARGET_OBJECTS:Base>
-               $<TARGET_OBJECTS:Clib>
-               $<TARGET_OBJECTS:Cont>
-               $<TARGET_OBJECTS:Foundation>
-               $<TARGET_OBJECTS:Lzma>
-               $<TARGET_OBJECTS:Lz4>
-               $<TARGET_OBJECTS:Zstd>
-               $<TARGET_OBJECTS:Zip>
-               $<TARGET_OBJECTS:Meta>
-               $<TARGET_OBJECTS:GuiCore>
-               $<TARGET_OBJECTS:TextInput>
-               ${macosx_objects}
-               ${unix_objects}
-               ${winnt_objects})
-
-ROOT_OBJECT_LIBRARY(BaseTROOT ${CMAKE_SOURCE_DIR}/core/base/src/TROOT.cxx ${CMAKE_BINARY_DIR}/ginclude/RGitCommit.h)
-target_include_directories(BaseTROOT PRIVATE
-   ${CMAKE_SOURCE_DIR}/core/foundation/res
-   ${CMAKE_SOURCE_DIR}/core/base/inc
-   ${CMAKE_SOURCE_DIR}/core/clib/inc
-   ${CMAKE_SOURCE_DIR}/core/meta/inc
-   ${CMAKE_SOURCE_DIR}/core/cont/inc
-   ${CMAKE_SOURCE_DIR}/core/foundation/inc
-   ${CMAKE_SOURCE_DIR}/core/thread/inc
-   ${CMAKE_SOURCE_DIR}/core/gui/inc
-   ${CMAKE_BINARY_DIR}/ginclude
-)
```

Comment:
They now come automatically from the `Core` target, added from each subdirectory where it makes sense (i.e. `core/cont/inc` added by `core/cont/CMakeLists.txt` to the `Core` target).

Reply:
All intermediate states should work, I developed this gradually. When I removed some includes, it's because they are no longer necessary. Note that above the whole object library target is being removed, not just the includes. I am adding the source file directly to the `Core` target.

Reply:
Actually, I forgot about the reworking on top of the new master, I will have to rebase or we should use a merge commit for this to ensure it's easy to revert if needed.